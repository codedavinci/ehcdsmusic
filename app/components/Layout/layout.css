.main-container {
  display: flex;
  flex-direction: column;
}

.app-header {
  display: flex;
  flex-direction: column;
  background-image: linear-gradient(to left, #434343 0%, black 100%);
  width: 100%;
  height: 50px;
  padding: 20px;
  border-bottom: 1px solid lightgray;
}
