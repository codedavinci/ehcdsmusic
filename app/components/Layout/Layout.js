import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import logo from '../../Sproutify.png'
import './layout.css'

const Logo = ({ handleDefault }) => (
  <div>
    <Link to="/" onClick={handleDefault}>
      <img src={logo} alt="That's the logo" className="app-logo" />
    </Link>
  </div>
)

const Layout = ({ children, handleDefault }) => (
  <div className="main-container">
    <div className="app-header">
      <Logo handleDefault={handleDefault} />
    </div>
    {children}
  </div>
)

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
