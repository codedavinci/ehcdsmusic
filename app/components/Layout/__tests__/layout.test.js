import React from 'react'

import Layout from '../Layout'

describe('<Layout />', () => {
  const defaultProps = { handleDefault: jest.fn() }

  const doShallow = (overrides = {}) => {
    const layout = shallow(<Layout {...overrides}>children</Layout>)
    return layout
  }
  it('renders', () => {
    const layout = doShallow(defaultProps)
    expect(layout).toMatchSnapshot()
  })

  it('should contain a full width scale - viewport', () => {})
  it('should render a full width header', () => {})
  it('should not render st', () => {})
})
