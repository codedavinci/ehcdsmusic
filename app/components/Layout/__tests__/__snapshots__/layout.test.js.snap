// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders 1`] = `
<div
  className="main-container"
>
  <div
    className="app-header"
  >
    <Logo
      handleDefault={[MockFunction]}
    />
  </div>
  children
</div>
`;
